### PR TITLE
Update for Ruby Saml 1.0 [US659]

### PR DESCRIPTION
**Pull request by sthanson**
* validate! was removed in version 1.0, so updated to is_valid?
* the saml_settings must be passed now in Response.new in order to decrypt data data, so changed the order.
* All of our code expects values (not arrays), but some values in 1.0 are wrapped in arrays... so wrap every value in an array and then take the first 
https://rally1.rallydev.com/#/30710832703d/detail/task/38574094196

### CODE REVIEW

File: lib/omniauth/strategies/saml.rb [hunk 1]
```diff
@@ -39,18 +39,19 @@ def callback_phase
...
-        @attributes = response.attributes
+        @attributes = {}
+        response.attributes.each{|key, val| @attributes[key.to_sym] = Array.wrap(val).first }
 
         if @name_id.nil? || @name_id.empty?
           raise OmniAuth::Strategies::SAML::ValidationError.new("SAML response missing 'name_id'")
         end
 
-        response.validate!
+        response.is_valid?
```
Comment (n3bulous) [inline lib/omniauth/strategies/saml.rb]: Does is_valid? raise an exception?  This looks like an unused value.

Comment (sthanson) [inline lib/omniauth/strategies/saml.rb]: @n3bulous This strategy passes exceptions onto Omniauth, which handles them.  (Omniauth turns the error into a flash message and then redirects the user to the #failure endpoint.

**Comment by n3bulous** [comment]
:+1:

**Comment by charlieok** [comment]
:+1:

**Pull request merged**